pypdf/_reader.py

```diff
@@ -776,9 +808,9 @@ def _read_standard_xref_table(self, stream: StreamType) -> None:
     def _read_xref_tables_and_trailers(
         self, stream: StreamType, startxref: Optional[int], xref_issue_nr: int
     ) -> None:
-        self.xref: Dict[int, Dict[Any, Any]] = {}
-        self.xref_free_entry: Dict[int, Dict[Any, Any]] = {}
-        self.xref_objStm: Dict[int, Tuple[Any, Any]] = {}
+        self.xref = {}
+        self.xref_free_entry = {}
+        self.xref_objStm = {}
```

Comment:
unless I'm wrong these are now in __init__ and no more required

Reply:
It seems like the intended use-case is to clean up these references on exit explicitly to free possible resources. I am not sure if there really are any benefits when the reader itself could be garbage collected at any point in time afterwards as well.

Reply:
You are right @stefan6419846. However, the garbage collection can happen any time. It's more of a guideline/best practice to clean up, and it's now somewhat consistent with other parts of the code where `close()` is used as well. I don't have a strong preference though, up to you/maintainers to decide.